state-sync/diem-data-client/src/diemnet/metrics.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_crypto::_once_cell::sync::Lazy;
+use diem_metrics::{
+    register_histogram_vec, register_int_counter_vec, HistogramTimer, HistogramVec, IntCounterVec,
+};
+
+pub const TOTAL: &str = "TOTAL";
+
+/// Counter for tracking sent requests
+pub static SENT_REQUESTS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_sent_requests",
+        "Counters related to sent requests",
+        &["request_types"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking success responses
+pub static SUCCESS_RESPONSES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_success_responses",
+        "Counters related to success responses",
+        &["response_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking error responses
+pub static ERROR_RESPONSES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_error_responses",
+        "Counters related to error responses",
+        &["response_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking peer selection errors
+pub static PEER_SELECTION_ERRORS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_data_client_peer_selection_errors",
+        "Counter for tracking peer selection errors",
+        &["response_type"]
+    )
+    .unwrap()
+});
+
+/// Counter for tracking request latencies
+pub static REQUEST_LATENCIES: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
+        "diem_data_client_request_latencies",
+        "Counters related to request latencies",
+        &["request_type"]
+    )
+    .unwrap()
+});
+
+/// Increments the given counter with the provided label values.
+/// The special label TOTAL stores the sum of all values in the counter.
```

Comment:
I'm not sure if this is a common way to do this. (Alternatively, I can have a separate counter for total (e.g., `REQUEST_COUNT`, `RESPONSE_COUNT`, etc.), but that would add more variable names and might be confusing). Please feel free to suggest another way.

Reply:
Yeah, this is one way of doing it, and I think it's fine. Another way would be dynamically aggregate across all labels on the visualization side (i.e., add them all up to get the total), meaning 1 less label to deal with, but then it becomes a client-side responsibility. I think this is okay.